packages/schema-type/.npmignore

```diff
@@ -0,0 +1,6 @@
+*
+
+!dist/**
```

Comment:
with the current config in `package.json` the `main` field would always be set on the ts version, never the dist version. Do we want that behaviour here? In other ts packages we use this yarn script to change the `main` field to `dist/index.js` before publishing: https://github.com/latticexyz/mud/blob/1f2598cff40cd9f5059b553b9291ffd2c61bacdd/packages/recs/package.json#L19-L20

But it might be that this package is intended to be only used in typescript since `enum` is a typescript feature anyway

Reply:
I actually tried out different things for `main`. Ultimately I kept it as typescript by default since it's flagship feature is an enum, but also include `dist` just in case someone needs `.js` stuff.
I imagine we can try it out like this and change later if necessary, unless you think `.ts` main is bad? I don't have much experience here, and locally it works either way

Reply:
I agree with this reasoning, let's try it out as is and later change if needed

Reply:
the pattern I've seen is building `.js` (for `main`) and `.d.ts` (for `types`):
https://github.com/wagmi-dev/wagmi/blob/main/packages/core/package.json#L27-L28

but that probably only makes sense if there are things in this package useful in JS alone (not TS)